### PR TITLE
Workflow landing improvements

### PR DESCRIPTION
**Pull request by mvdbeek**
Optimizing the BRC / IWC case first here:
- Separate creation of public and private landing requests
- Don't claim the landing page if public=true is passed in request
- Allow passing in a TRS url so we don't need to hardcode instance-specific workflow ids
- Don't reimport TRS workflow if the user already has it
- Redirect user to login / registration form if not logged in, then send them back to landing page

Also adds a selenium test that exercises public and private landing requests.  

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 1]
```diff
@@ -1171,11 +1171,7 @@ def create_landing(
         trans: ProvidesUserContext = DependsOnTrans,
         workflow_landing_request: CreateWorkflowLandingRequestPayload = Body(...),
     ) -> WorkflowLandingRequest:
-        try:
-            return self.landing_manager.create_workflow_landing_request(workflow_landing_request)
-        except Exception:
-            log.exception("Problem...")
-            raise
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/workflows.py:1178]: Opps...

File: test/unit/app/managers/test_landing.py [hunk 2]
```diff
@@ -64,7 +71,9 @@ def test_tool_landing_requests_requires_matching_client_secret(self):
         assert exception is not None
 
     def test_tool_landing_requests_get_requires_claim(self):
-        landing_request: ToolLandingRequest = self.landing_manager.create_tool_landing_request(self._tool_request)
```
Comment (jmchilton) [inline test/unit/app/managers/test_landing.py]: Shouldn't this be a separate test for a second path through the code?

File: client/src/components/Landing/WorkflowLanding.vue [hunk 3]
```diff
@@ -21,21 +20,17 @@ const props = withDefaults(defineProps<Props>(), {
...
 const requestState = ref<Record<string, never> | null>(null);
+const instance = ref<boolean>(false);
 
-onMounted(async () => {
-    const payload: ClaimLandingPayload = {};
-    if (props.secret) {
-        payload["client_secret"] = props.secret;
-    }
-
-    const { data, error } = await GalaxyApi().POST("/api/workflow_landings/{uuid}/claim", {
```
Comment (jmchilton) [inline client/src/components/Landing/WorkflowLanding.vue:31]: I mean... the intended approach should still work right 😅 . You did something very different here it seems and that is fine but I think private claims should still work.

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Fair, I can create separate mount points

Comment (jmchilton) [reply client/src/components/Landing/WorkflowLanding.vue:31]: xref https://github.com/mvdbeek/iwc/blob/website/website/pages/workflow/%5Bid%5D.vue#L56-L73 - is the idea here that it works with an already logged in Galaxy only and the request is made as the user? Why not give a link that requires a log in when you land on it? Is that hard to implement?

Update: I should have asked that when you were looking at CORS stuff - sorry I didn't put the pieces together sooner.

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: > xref https://github.com/mvdbeek/iwc/blob/website/website/pages/workflow/%5Bid%5D.vue#L56-L73 - is the idea here that it works with an already logged in Galaxy only and the request is made as the user? 

No, the landing page creation can only be anonymous as far as I understand. Calls from external sites shouldn't send sessions or similar sensitive stuff.

> Why not give a link that requires a log in when you land on it?

That's what I'm doing when you arrive at the landing page ... right now you have to be logged in but the idea is to just embed the login / register component and move on once they're logged in.

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Or are you asking why I'm not claiming the landing request ? That's just cause navigating back, refreshing, and sharing should work IMO.

Comment (jmchilton) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Will sharing work - you're creating the landing with a particular user in there - I feel like I had tests to ensure that is pretty explicitly tied to the user. I would call that a serious bug/oversight on my part if it works. Could we introduce something in the model to declare that landing is anonymous and works that way? I'd be happy to take over the PR and implement that - you already have a nice migration setup in the PR and I don't want to create more work for you.

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Sorry John, I don't see how the landing would be useful for BRC if we need an authenticated user. I don't think it's a bug. Isn't the client secret the way to make it private ?

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Discussed this on matrix and I think we're on the same page now, we need to separate private and public workflow landing requests

Comment (mvdbeek) [reply client/src/components/Landing/WorkflowLanding.vue:31]: Alright, both is possible now depending on the value of the `public` query parameter. If not provided we try to claim the request.

File: test/unit/app/managers/test_landing.py (hunk 2 above)
Comment (mvdbeek) [reply test/unit/app/managers/test_landing.py]: I think this is resolved by making the public/private distinction explicit ? I'm catching `ItemMustBeClaimed` here now.

File: client/src/components/Workflow/Run/services.js [hunk 4]
```diff
@@ -12,8 +12,8 @@ import { rethrowSimple } from "utils/simple-error";
  * @param {String} workflowId - (Stored?) Workflow ID to fetch data for.
  * @param {String} version - Version of the workflow to fetch.
  */
-export async function getRunData(workflowId, version = null) {
-    let url = `${getAppRoot()}api/workflows/${workflowId}/download?style=run`;
+export async function getRunData(workflowId, version = null, instance = false) {
+    let url = `${getAppRoot()}api/workflows/${workflowId}/download?style=run&instance=${instance}`;
```
Comment (jmchilton) [inline client/src/components/Workflow/Run/services.js:16]: Fetching by instance ID gave me really unexpected behavior in workflow rerun API - I'd check that out and make sure this is giving you the expected behavior and that you and I are the on the same page about what that expected behavior should be in this case. The PR I'm referring to is https://github.com/galaxyproject/galaxy/pull/18985 and the commit that fixed the API behavior IMO is https://github.com/galaxyproject/galaxy/pull/18985/commits/cc8f1b65b8da880d1b22135eea39220c3198b9c0 - you may need to apply the same fix to this endpoint if we're on the same page.

**Comment by jmchilton** [comment]
Thanks for making those changes - I really appreciate it. The API makes a lot of coherent sense to me now but I hate creating extra work for you. Also I love seeing Selenium tests - thanks so much for doing that!

**Comment by jmchilton** [comment]
I never see those tests fail unless I have broken something but there isn't logging, it won't let me rerun them, and I have no clue how any of those changes could break those tests - you haven't touched tool parsing code here - so I am going to assume those spurious failures I just haven't seen before.

**Pull request merged**